include/fuzzuf/coverage/shm_coverage.hpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * fuzzuf
+ * Copyright (C) 2022 Ricerca Security
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+/**
+ * @file shm_coverage.hpp
+ * @author Ricerca Security <fuzzuf-dev@ricsec.co.jp>
+ */
+
+#ifndef FUZZUF_INCLUDE_COVERAGE_SHM_COVERAGE_HPP
+#define FUZZUF_INCLUDE_COVERAGE_SHM_COVERAGE_HPP
+
+#include "fuzzuf/feedback/inplace_memory_feedback.hpp"
+#include "fuzzuf/logger/logger.hpp"
+#include "fuzzuf/utils/common.hpp"
+
+/**
+ * @class ShmCoverage
+ * @brief Base class that represents coverage stored in shared memory.
+ */
+class ShmCoverage {
```

Comment:
I feel naming this kind of classes `FoobarCoverage` is a bit ambiguous. What this class does is setting up shared memory and maintaining it to let other classes obtain coverage information. In other words, this class does not coverage itself, and there may be other more comprehensible names.

Reply:
Say, something like `ShmCoverageMaintainer`? Nope this sounds lame and dumb.
Anyways any suggestion for another name?

Reply:
I agree. These classes need to have a suffix to represent their role. The candidates are: `Manager`, `Provider`, `Controller`. I prefer to use `Manager` because it is widely used, but I'm afraid that it is unclear since this word is often overused.

Reply:
Yeah I have the same concern. It would be preferable if there is a more concrete name.
How about `ShmCoverageAttacher`? I'm imagining the class to attach new shared memory to executors for adding an extra feedback.

Reply:
:+1: to ShmCoverageAttacher.
Once I thought of `ShmCoverageDispatcher`, yet its inappropriate since Coverage itself is not a function.

Reply:
Thanks for the comment. I'll add changes to use `Attacher` for the classes.

Reply:
Applied with [cc2c909](https://github.com/fuzzuf/fuzzuf/pull/34/commits/cc2c90959118fee39622228c96aba8f62c43664c).